okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java

```diff
@@ -132,6 +135,34 @@ public AndroidPlatform(Class<?> sslParametersClass, OptionalMethod<Socket> setUs
     }
   }
 
+  @Override public Object captureAllocationSite() {
+    if (missingCloseGuard.get()) return null;
+
+    try {
+      Class<?> closeGuardClass = Class.forName("dalvik.system.CloseGuard");
```

Comment:
Agree. I opened an issue on AOSP to ask for a way to hook into this. I'm guessing it will get closed, but I figured it wouldn't hurt to ask.